### PR TITLE
Reporter can add organisation

### PR DESCRIPTION
**Pull request by dleard**
- Generalizes 'Add Facility' component to accomodate organisations
- adds createOrganisation mutation
- fixes incorrect primary key in ciip_user_organisation table

### CODE REVIEW

File: app/components/AddOrganisationFacility.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import FormFieldTemplate from 'containers/Forms/FormFieldTemplate';
+import FormObjectFieldTemplate from 'containers/Forms/FormObjectFieldTemplate';
+import addOperatorSchema from 'components/organisation/addOrganisationSchema';
+import addFacilitySchema from 'components/facility/addFacilitySchema';
+
+interface Props {
+  onAddOrganisation?: (...args: any[]) => void;
+  onAddFacility?: (...args: any[]) => void;
+  organisationRowId?: number;
+  reportingYear?: number;
```
Comment (matthieu-foucault) [inline app/components/AddOrganisationFacility.tsx]: Aren't all of those props required?

File: app/components/AddOrganisationFacility.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+const AddOrganisationFacility: React.FunctionComponent<Props> = props => {
+  const {
+    onAddOrganisation,
+    onAddFacility,
+    organisationRowId,
+    reportingYear
+  } = props;
+  const [modalShow, setModalShow] = useState(false);
```
Comment (matthieu-foucault) [inline app/components/AddOrganisationFacility.tsx]: Rename to `isModalVisible`

File: app/components/AddOrganisationFacility.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    const variables = {
+      input: {
+        facility: {
+          facilityName,
+          facilityType,
+          bcghgid,
+          naicsCode,
+          ...mailingAddress,
+          // TODO: Our facility table currently stores reportingYear as a string. Should it be an int? It's also a string in swrs & init functions
+          reportingYear: String(reportingYear),
```
Comment (matthieu-foucault) [inline app/components/AddOrganisationFacility.tsx]: I don't think that reportingYear should be a field in the facility table.

File: app/components/facility/addFacilitySchema.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import {JSONSchema6} from 'json-schema';
+
+const addFacilitySchema: JSONSchema6 = {
+  type: 'object',
+  title: 'Facility Details',
+  properties: {
```
Comment (matthieu-foucault) [inline app/components/facility/addFacilitySchema.ts]: I would only keep the facility name and BC GHG ID. The other information doesn't really need to be in the facility table

File: app/components/organisation/addOrganisationSchema.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
+import {JSONSchema6} from 'json-schema';
+
+const addOperatorSchema: JSONSchema6 = {
+  type: 'object',
+  title: 'Operator Details',
+  properties: {
+    operatorName: {
```
Comment (matthieu-foucault) [inline app/components/organisation/addOrganisationSchema.ts]: Similarly to the facility, I would only keep the operator name in here (and maybe the CRA number)

File: app/components/AddOrganisationFacility.tsx (hunk 1 above)
Comment (dleard) [reply app/components/AddOrganisationFacility.tsx]: They are required conditionally, if AddOrganisationFacility is being rendered from the FacilityList container it needs the bottom 3 props, if being rendered from the Organisations container it only needs the first prop.

**Pull request merged**